### ISSUE TITLE
Can't start Tribler from devel with twistd anymore

### ISSUE THREAD

**Issue by vi** [Tribler/tribler#2830]
##### Tribler version/branch+revision: #####

Git bisect shows a2437b03dd267bbec4c1a621628da0d9e991e99a

##### Operating system and version: #####

Linux

##### Relevant log file output: #####

```
2017-02-27T21:40:31+0300 [-] Tribler started
2017-02-27T21:40:32+0300 [-] object of type 'NoneType' has no len()
```

**Comment by devos50** [comment]
It seems that Tribler has correctly started. Can you check whether the API has booted by executing:

```
curl http://localhost:8085/downloads
```

?

**Comment by vi** [comment]
Twistd exits immediately. Fuller log:

```
2017-02-27T23:31:21+0300 [-] Yappi not installed, profiling options won't be available
2017-02-27T23:31:21+0300 [twisted.scripts._twistd_unix.UnixAppLogger#info] twistd 16.2.0 (/usr/bin/python 2.7.9) starting up.
2017-02-27T23:31:21+0300 [twisted.scripts._twistd_unix.UnixAppLogger#info] reactor class: twisted.internet.epollreactor.EPollReactor.
2017-02-27T23:31:21+0300 [-] Starting Tribler
2017-02-27T23:31:21+0300 [-] Site starting on 8085
2017-02-27T23:31:21+0300 [twisted.web.server.Site#info] Starting factory <twisted.web.server.Site instance at 0x7fce0ca31d40>
INFO    1488227481.92       community:307   initializing:  MultiChainCommunity
2017-02-27T23:31:21+0300 [-] Factory starting on 6639
2017-02-27T23:31:21+0300 [twisted.internet.protocol.Factory#info] Starting factory <twisted.internet.protocol.Factory instance at 0x7fce086557e8>
2017-02-27T23:31:21+0300 [-] Factory starting on 15021
2017-02-27T23:31:21+0300 [twisted.internet.protocol.Factory#info] Starting factory <twisted.internet.protocol.Factory instance at 0x7fce08655ab8>
2017-02-27T23:31:21+0300 [-] Factory starting on 46386
2017-02-27T23:31:21+0300 [twisted.internet.protocol.Factory#info] Starting factory <twisted.internet.protocol.Factory instance at 0x7fce08655cf8>
2017-02-27T23:31:21+0300 [-] Factory starting on 22067
2017-02-27T23:31:21+0300 [twisted.internet.protocol.Factory#info] Starting factory <twisted.internet.protocol.Factory instance at 0x7fce08655f38>
2017-02-27T23:31:21+0300 [-] Factory starting on 19839
2017-02-27T23:31:21+0300 [twisted.internet.protocol.Factory#info] Starting factory <twisted.internet.protocol.Factory instance at 0x7fce0866b1b8>
2017-02-27T23:31:22+0300 [-] Tribler started
2017-02-27T23:31:22+0300 [-] object of type 'NoneType' has no len()
2017-02-27T23:31:22+0300 [twisted.internet.protocol.Factory] (TCP Port 15021 Closed)
2017-02-27T23:31:22+0300 [twisted.internet.protocol.Factory#info] Stopping factory <twisted.internet.protocol.Factory instance at 0x7fce08655ab8>
2017-02-27T23:31:22+0300 [twisted.internet.protocol.Factory] (TCP Port 46386 Closed)
2017-02-27T23:31:22+0300 [twisted.internet.protocol.Factory#info] Stopping factory <twisted.internet.protocol.Factory instance at 0x7fce08655cf8>
2017-02-27T23:31:22+0300 [twisted.internet.protocol.Factory] (TCP Port 19839 Closed)
2017-02-27T23:31:22+0300 [twisted.internet.protocol.Factory#info] Stopping factory <twisted.internet.protocol.Factory instance at 0x7fce0866b1b8>
2017-02-27T23:31:22+0300 [twisted.web.server.Site] (TCP Port 8085 Closed)
2017-02-27T23:31:22+0300 [twisted.web.server.Site#info] Stopping factory <twisted.web.server.Site instance at 0x7fce0ca31d40>
2017-02-27T23:31:22+0300 [twisted.internet.protocol.Factory] (TCP Port 6639 Closed)
2017-02-27T23:31:22+0300 [twisted.internet.protocol.Factory#info] Stopping factory <twisted.internet.protocol.Factory instance at 0x7fce086557e8>
2017-02-27T23:31:22+0300 [twisted.internet.protocol.Factory] (TCP Port 22067 Closed)
2017-02-27T23:31:22+0300 [twisted.internet.protocol.Factory#info] Stopping factory <twisted.internet.protocol.Factory instance at 0x7fce08655f38>
```

`for((;;)) { curl http://localhost:8085/downloads; }` does not manage to connect.

**Comment by vi** [comment]
```
$ twistd -b -n tribler  
Removing stale pidfile /home/tribler/tribler/twistd.pid
2017-02-27T23:34:46+0300 [-] Yappi not installed, profiling options won't be available
2017-02-27T23:34:46+0300 [twisted.scripts._twistd_unix.UnixAppLogger#info] twistd 16.2.0 (/usr/bin/python 2.7.9) starting up.
2017-02-27T23:34:46+0300 [twisted.scripts._twistd_unix.UnixAppLogger#info] reactor class: twisted.internet.epollreactor.EPollReactor.
> /usr/lib/python2.7/dist-packages/twisted/internet/base.py(1193)run()
-> self.startRunning(installSignalHandlers=installSignalHandlers)
(Pdb) c
2017-02-27T23:34:54+0300 [-] Starting Tribler
2017-02-27T23:34:54+0300 [-] Site starting on 8085
2017-02-27T23:34:54+0300 [twisted.web.server.Site#info] Starting factory <twisted.web.server.Site instance at 0x7f66506bd320>
INFO    1488227694.58       community:307   initializing:  MultiChainCommunity
2017-02-27T23:34:54+0300 [-] Factory starting on 51963
2017-02-27T23:34:54+0300 [twisted.internet.protocol.Factory#info] Starting factory <twisted.internet.protocol.Factory instance at 0x7f664c1cb488>
2017-02-27T23:34:54+0300 [-] Factory starting on 46949
2017-02-27T23:34:54+0300 [twisted.internet.protocol.Factory#info] Starting factory <twisted.internet.protocol.Factory instance at 0x7f664c1cb8c0>
2017-02-27T23:34:54+0300 [-] Factory starting on 17022
2017-02-27T23:34:54+0300 [twisted.internet.protocol.Factory#info] Starting factory <twisted.internet.protocol.Factory instance at 0x7f664c1cbb00>
2017-02-27T23:34:54+0300 [-] Factory starting on 5941
2017-02-27T23:34:54+0300 [twisted.internet.protocol.Factory#info] Starting factory <twisted.internet.protocol.Factory instance at 0x7f664c1cbd40>
2017-02-27T23:34:54+0300 [-] Factory starting on 42031
2017-02-27T23:34:54+0300 [twisted.internet.protocol.Factory#info] Starting factory <twisted.internet.protocol.Factory instance at 0x7f664c1cbf80>
2017-02-27T23:34:54+0300 [-] Tribler started
Jumping into debugger for post-mortem of exception 'object of type 'NoneType' has no len()':
> /home/tribler/tribler/Tribler/Core/APIImplementation/LaunchManyCore.py(657)resume_download()
-> 'len %d ' % len(pstate.get('state', 'engineresumedata')) if has_resume_data else 'None')
(Pdb) bt
  /usr/lib/python2.7/dist-packages/twisted/internet/defer.py(588)_runCallbacks()
-> current.result = callback(current.result, *args, **kw)
  /home/tribler/tribler/Tribler/Core/Session.py(562)load_checkpoint()
-> self.load_checkpoint()
  /home/tribler/tribler/Tribler/Core/Session.py(526)load_checkpoint()
-> self.lm.load_checkpoint()
  /home/tribler/tribler/Tribler/Core/APIImplementation/LaunchManyCore.py(598)load_checkpoint()
-> do_load_checkpoint()
  /home/tribler/tribler/Tribler/Core/APIImplementation/LaunchManyCore.py(595)do_load_checkpoint()
-> self.resume_download(filename, setupDelay=i * 0.1)
> /home/tribler/tribler/Tribler/Core/APIImplementation/LaunchManyCore.py(657)resume_download()
-> 'len %d ' % len(pstate.get('state', 'engineresumedata')) if has_resume_data else 'None')
(Pdb) 
```

**Comment by vi** [comment]
With the following hack it starts and works:

```diff
diff --git a/Tribler/Core/APIImplementation/LaunchManyCore.py b/Tribler/Core/APIImplementation/LaunchManyCore.py
index 979023c16..d8a5d2464 100644
--- a/Tribler/Core/APIImplementation/LaunchManyCore.py
+++ b/Tribler/Core/APIImplementation/LaunchManyCore.py
@@ -653,8 +653,9 @@ class TriblerLaunchMany(TaskManager):
 
         if pstate is not None:
             has_resume_data = pstate.get('state', 'engineresumedata') is None
+            erd = pstate.get('state', 'engineresumedata')
             self._logger.debug("tlm: load_checkpoint: resumedata %s",
-                               'len %d ' % len(pstate.get('state', 'engineresumedata')) if has_resume_data else 'None')
+                               'len %d ' % len(erd) if has_resume_data and erd is not None else 'None')
 
         if tdef and dscfg:
             if dscfg.get_dest_dir() != '':  # removed torrent ignoring
```

**Comment by devos50** [comment]
Oh, that's a silly bug introduced by my :/, will be fixed in #2833.
